nixos/modules/services/monitoring/osquery.nix

```diff
@@ -0,0 +1,114 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+let
+  cfg = config.services.osquery;
+  logsDirectoryPrefix = "/var/log/";
+  stateDirectoryPrefix = "/var/lib/";
+  runtimeDir = "/run/osquery";
+  dirname = path: with lib.strings; with lib.lists; concatStringsSep "/"
+    (init (splitString "/" (normalizePath path)));
+
+  # conf is the osquery configuration file used when the --config_plugin=filesystem.
+  # filesystem is the osquery default value for the config_plugin flag.
+  conf = pkgs.writeText "osquery.conf" (builtins.toJSON cfg.settings);
+
+  # flagfile is the file containing osquery command line flags to be
+  # provided to the application using the special --flagfile option.
+  flagfile = pkgs.writeText "osquery.flags"
+    (concatStringsSep "\n"
+      (mapAttrsToList (name: value: "--${name}=${value}")
+        # Use the conf derivation if not otherwise specified.
+        ({ config_path = conf; } // cfg.flags)));
+
+  osqueryi = pkgs.runCommand "osqueryi" { nativeBuildInputs = [ pkgs.makeWrapper ]; } ''
+    mkdir -p $out/bin
+    ln -s ${pkgs.osquery}/bin/osqueryi $out/bin/osqueryi
+    wrapProgram $out/bin/osqueryi \
+      --add-flags "--flagfile ${flagfile}"
```

Comment:
Can't we move this to the package and make it use a env variable?

Reply:
Yes we could but what would be the benefit? 
If needed, this could be moved later without impacting existing users. So, i would prefer to avoid doing it in this PR ;)
 
(Also, I'm not sure modifying the behavior of the binary in the osquery package is a good practice.)

Reply:
We would be avoiding an extra package and make the package easier to use outside of the module

Reply:
But IIUC, i would then have to wrap the osquery package here to set the environment variable. Is it correct?
So, outside of the module, i don't see how it is easier to type
```
ENV=path/to/flagfile osqueryi
``` 
than
```
osqueryi --flagfile path/to/flagfile
```
Moreoever, it wouldn't be documented.

hm, I should be missing something :thinking:

Reply:
I agree with @nlewo here, but there should be a way to make available the modified osquery for local administrators or something so they don't have to figure where is the correct flagfile to pass.

Reply:
`osqueryi` is already correctly wired up.

Reply:
@RaitoBezarius the current wrapper already allow admin to use osqueryi without having to know where the flag file is.